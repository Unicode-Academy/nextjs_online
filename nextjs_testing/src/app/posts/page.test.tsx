import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import PostPage from "./page";
describe("Posts", () => {
  it("renders a heading", async () => {
    global.fetch = jest.fn().mockResolvedValue({
      json: jest.fn().mockResolvedValue([
        {
          id: 1,
          title: "Learn ReactJS",
          body: "Detail Learn ReactJS",
        },

        {
          id: 2,
          title: "Learn NextJS",
          body: "Detail Learn NextJS",
        },
      ]),
    });

    const ResolvePage = await Promise.resolve(PostPage());
    render(ResolvePage);

    const heading = screen.getByRole("heading", { level: 1 });

    expect(heading).toContainHTML("Posts");

    const postTitleList = screen.getAllByTestId("post-title");
    const postTitleMap = postTitleList.map((item) => item.textContent);
    expect(postTitleMap).toEqual(["Learn ReactJS", "Learn NextJS"]);
  });

  it("render posts page unchanged", async () => {
    const ResolvePage = await Promise.resolve(PostPage());
    const { container } = render(ResolvePage);

    expect(container).toMatchSnapshot();
  });
});
