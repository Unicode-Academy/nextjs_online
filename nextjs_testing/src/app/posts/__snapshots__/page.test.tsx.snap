// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Posts render posts page unchanged 1`] = `
<div>
  <div>
    <h1>
      Posts
    </h1>
    <h2>
      Ahihi
    </h2>
    <div>
      <h2
        data-testid="post-title"
      >
        Learn ReactJS
      </h2>
      <p
        data-testid="post-body"
      >
        Detail Learn ReactJS
      </p>
    </div>
    <div>
      <h2
        data-testid="post-title"
      >
        Learn NextJS
      </h2>
      <p
        data-testid="post-body"
      >
        Detail Learn NextJS
      </p>
    </div>
  </div>
</div>
`;
